mesh2hrtf/Mesh2Input/mesh2input.py

```diff
@@ -1160,6 +1175,48 @@ def _render_pictures(filepath1, unitFactor):
         nameRender = ("%d_deg_azimuth" % azim[ii])
         bpy.data.images['Render Result'].save_render(
             os.path.join(dirRender, "%s.png" % nameRender))
+        
+
+#Added by Sanket
+def _write_checkall_txt(filepath1):
+    """check all before export"""
+    #check for geometry is solid
+    bpy.ops.mesh.print3d_check_solid()
+    info = report.info()
+    non_manifold = info[0][0]
+    non_manifold_num = int(re.search(r'\d+', non_manifold).group())
+    bad_contiguous = info[1][0]
+    bad_contiguous_num = int(re.search(r'\d+', bad_contiguous).group())
+    #check geometry for self intersections
+    bpy.ops.mesh.print3d_check_intersect()
+    info = report.info()
+    intersect_face = info[0][0]
+    intersect_face_num = int(re.search(r'\d+', intersect_face).group())
+    #check edges are below the sharpness preference
+    bpy.ops.mesh.print3d_check_sharp()
+    info = report.info()
+    sharp_edge = info[0][0]
+    sharp_edge_num = int(re.search(r'\d+', sharp_edge).group())
+
+    #If there is non-zero number after check-all then raise an error
+    if non_manifold_num != 0 or bad_contiguous_num != 0 or intersect_face_num != 0 or sharp_edge_num != 0:
+        file = open(os.path.join(filepath1, "Error_Checkall_info.txt"), "w",
+                    encoding="utf8", newline="\n")
```

Comment:
Any file operations should use the `with` statement. It makes sure that the file access is closed, even if an error occurs:
```suggestion
        with open(os.path.join(filepath1, "Error_Checkall_info.txt"), "w",
                    encoding="utf8", newline="\n") as file:
            file.write('text\n')
```

Lets also think about a good filename.